mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetListFragment.java

```diff
@@ -412,49 +412,49 @@ public void processContent(String responseString, boolean longPolling) {
         openHABWidgetAdapter.stopVideoWidgets();
         openHABWidgetAdapter.stopImageRefresh();
         // If openHAB verion = 1 get page from XML
-        if (mActivity.getOpenHABVersion() == 1) {
-            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
-            try {
-                DocumentBuilder builder = dbf.newDocumentBuilder();
-                Document document = builder.parse(new InputSource(new StringReader(responseString)));
-                if (document != null) {
-                    Node rootNode = document.getFirstChild();
-                    openHABWidgetDataSource.setSourceNode(rootNode);
+        if (responseString != null && responseString.length() > 0) {
```

Comment:
If the response string is null or empty what happens?  Line 433 handles the case where the xml document is null and I believe tries to reconnect. should something similar happen here as well?